connector-email/src/main/java/com/redhat/cloud/notifications/connector/email/EmailManagementProcessor.java

```diff
@@ -0,0 +1,112 @@
+
+        if (recipientsList.isEmpty()) {
+            Log.infof("Skipped Email notification because the recipients list was empty [orgId=$%s, historyId=%s]", exchange.getProperty(ORG_ID, String.class), exchange.getProperty(ID, String.class));
+        } else {
+            // send to bop
+            sendToBop(exchange, recipientsList);
+        }
+    }
+
+    private void sendToBop(Exchange exchange, Set<String> recipientsList) {
+        // split recipient list and send it to BOP
+        List<List<String>> packedRecipients = partition(recipientsList, emailConnectorConfig.getMaxRecipientsPerEmail() - 1);
+        final String subject = exchange.getProperty(ExchangeProperty.RENDERED_SUBJECT, String.class);
+        final String body = exchange.getProperty(ExchangeProperty.RENDERED_BODY, String.class);
+        final String sender = exchange.getProperty(ExchangeProperty.EMAIL_SENDER, String.class);
+
+        for (int i = 0; i < packedRecipients.size(); i++) {
+            final Timer.Sample bopResponseTimeMetric = Timer.start(meterRegistry);
+            bopManager.sendToBop(packedRecipients.get(i), subject, body, sender);
+            bopResponseTimeMetric.stop(meterRegistry.timer(BOP_RESPONSE_TIME_METRIC));
+            Log.infof("Sent Email notification %d/%d [orgId=%s, historyId=%s]", i + 1, packedRecipients.size(), exchange.getProperty(ORG_ID, String.class), exchange.getProperty(ID, String.class));
+        }
+    }
+
+    private static List<List<String>> partition(Set<String> collection, int n) {
+        AtomicInteger counter = new AtomicInteger();
+        return collection.stream()
+            .collect(Collectors.groupingBy(it -> counter.getAndIncrement() / n))
+            .values().stream().toList();
+    }
+
+    private Set<String> fetchRecipients(Exchange exchange) {
+        List<RecipientSettings> recipientSettings = exchange.getProperty(ExchangeProperty.RECIPIENT_SETTINGS, List.class);
+        Set<String> subscribers = exchange.getProperty(ExchangeProperty.SUBSCRIBERS, Set.class);
+        Set<String> unsubscribers = exchange.getProperty(ExchangeProperty.UNSUBSCRIBERS, Set.class);
+        JsonObject recipientsAuthorizationCriterion = exchange.getProperty(ExchangeProperty.RECIPIENTS_AUTHORIZATION_CRITERION, JsonObject.class);
+
+        boolean subscribedByDefault = exchange.getProperty(ExchangeProperty.SUBSCRIBED_BY_DEFAULT, boolean.class);
+        final String orgId = exchange.getProperty(ORG_ID, String.class);
+
+        final Timer.Sample recipientsResolverResponseTimeMetric = Timer.start(meterRegistry);
+        Set<String> recipientsList = externalRecipientsResolver.recipientUsers(
+                orgId,
+                Set.copyOf(recipientSettings),
+                subscribers,
+                unsubscribers,
+                subscribedByDefault,
+                recipientsAuthorizationCriterion)
+            .stream().map(User::getEmail).filter(email -> email != null && !email.isBlank()).collect(toSet());
```

Comment:
Same question here: I guess the errors will be handled by the exception processor, right?

Reply:
Yes it will